File: src/management/client/src/models/Signal.ts
// -----------------------------------------------------------------------
// <copyright file="Signal.ts" company="Microsoft Corporation">
//        Copyright (c) Microsoft Corporation.  All rights reserved.
// </copyright>
// -----------------------------------------------------------------------

import SignalConfiguration from './SignalConfiguration';
import ResourceType from '../enums/ResourceType';

/**
 * This model represents the Signal entity
 */
export default interface Signal {
    /**
     * Gets the signal id.
     */
    id: string;

    /**
     * Gets the signal name.
     */
    name: string;

    /**
     * Gets the signal supported cadences (in minutes).
     */
    supportedCadences: number[];

    /**
     * Gets the signal supported resource types.
     */
    supportedResourceTypes: ResourceType[];

    /**
     * Gets the signal configurations.
     */
    configurations: ReadonlyArray<SignalConfiguration>;
}

File: src/management/server/SmartSignalsManagementApi/Models/Signal.cs
﻿//-----------------------------------------------------------------------
// <copyright file="Signal.cs" company="Microsoft Corporation">
//        Copyright (c) Microsoft Corporation.  All rights reserved.
// </copyright>
//-----------------------------------------------------------------------

namespace Microsoft.Azure.Monitoring.SmartAlerts.Appliance.ManagementApi.Models
{
    using System.Collections.Generic;
    using Newtonsoft.Json;

    /// <summary>
    /// This class represents a signal model for the Management API responses.
    /// </summary>
    public class Signal
    {
        /// <summary>
        /// Gets or sets the signal id.
        /// </summary>
        [JsonProperty("id")]
        public string Id { get; set; }

        /// <summary>
        /// Gets or sets the signal name.
        /// </summary>
        [JsonProperty("name")]
        public string Name { get; set; }

        /// <summary>
        /// Gets or sets the signal supported cadences (in minutes).
        /// </summary>
        [JsonProperty("supportedCadences")]
        public List<int> SupportedCadences { get; set; }

        /// <summary>
        /// Gets or sets the signal supported resource types.
        /// </summary>
        [JsonProperty("supportedResourceTypes")]
        public List<ResourceType> SupportedResourceTypes { get; set; }

        /// <summary>
        /// Gets or sets the signal configurations.
        /// </summary>
        [JsonProperty("configurations")]
        public List<SignalConfiguration> Configurations { get; set; }
    }
}
